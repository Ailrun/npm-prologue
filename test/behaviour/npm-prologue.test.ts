import { ChildProcess, spawn } from 'child_process';
import { join } from 'path';
import { promisify } from 'util';

import rimraf = require('rimraf');

import { exitCode, packageRoot, stdoutData } from './test-utils';

const execPath = join(packageRoot, 'dist/npm-prologue.js');

describe('npm-prolgue', () => {
  let npmPrologue: ChildProcess;

  afterEach(async () => {
    npmPrologue.kill();
    await promisify(rimraf)(join(__dirname, '__temp__/*'));
  });

  it('should be able to run with normally', async () => {
    npmPrologue = spawn(execPath, [], {
      stdio: 'pipe',
      cwd: join(__dirname, '/__temp__'),
    });
    const logo = await stdoutData(npmPrologue.stdout);
    expect(logo).toMatchSnapshot();

    const code = await exitCode(npmPrologue);
    expect(code).toBe(0);
  });
});
