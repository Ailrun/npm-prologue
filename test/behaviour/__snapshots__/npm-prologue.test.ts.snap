// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npm-prolgue should be able to run with normally 1`] = `
"                                                     _                            
  _ __   _ __   _ __ ___          _ __   _ __  ___  | |  ___    __ _  _   _   ___ 
 | '_ \\\\ | '_ \\\\ | '_ \` _ \\\\  _____ | '_ \\\\ | '__|/ _ \\\\ | | / _ \\\\  / _\` || | | | / _ \\\\
 | | | || |_) || | | | | ||_____|| |_) || |  | (_) || || (_) || (_| || |_| ||  __/
 |_| |_|| .__/ |_| |_| |_|       | .__/ |_|   \\\\___/ |_| \\\\___/  \\\\__, | \\\\__,_| \\\\___|
        |_|                      |_|                           |___/              
"
`;
